runtime/src/consensus/state/beacon.rs

```diff
@@ -87,6 +89,26 @@ impl MutableState {
     }
 }
 
+/// Consensus beacon state.
```

Comment:
What is the difference between this and `ImmutableState`? One is for low-level state access and the other one is a more high-level API? Why not just merge the two, to avoid the need to refactor other parts?

Reply:
Wanted to keep `ImmutableState` clean as possible (only low lever operations). Will merge.